Automattic/wp-calypso
# composite-checkout: Add p24 payment method

#### Changes proposed in this Pull Request

This adds the p24 payment method to composite checkout.

#### Screenshots

Old checkout:

![Screen Shot 2020-06-26 at 5 25 30 PM](https://user-images.githubusercontent.com/2036909/85904498-e4ef6500-b7d6-11ea-9f50-6168eb59ae3f.png)

New checkout (closed):

![Screen Shot 2020-06-26 at 5 47 46 PM](https://user-images.githubusercontent.com/2036909/85904508-ea4caf80-b7d6-11ea-981a-d3d1ae003a19.png)

New checkout (open):

![Screen Shot 2020-06-26 at 5 59 24 PM](https://user-images.githubusercontent.com/2036909/85904517-ef116380-b7d6-11ea-9f2f-3fcb85ab5dfd.png)


#### Testing instructions

- Apply D45583-code to force payment method to be available.
- Sandbox the store.
- Change your user's currency to `PLN`.
- Add a plan to your cart and visit checkout. 
- Add `?flags=composite-checkout-force` to the query string in the URL and reload the page because currently it is disabled for international currencies.
- Verify that "P24" is an option displayed in the final payment step.
- Verify that you cannot pay if the "Name" field is empty.
- Fill out the fields and submit the payment.
- Verify that you are redirected to a Stripe test page.
- Click to accept the payment and verify that the payment completes successfully and that the plan is purchased.
